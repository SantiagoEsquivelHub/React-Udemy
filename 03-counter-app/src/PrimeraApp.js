/* import React, { Fragment } from 'react'; */
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import './index.css';

//Functional Components

const PrimeraApp = ({saludo, subtitulo}) => {

    //const saludo = 'Hola mundo';


    return (
        /* Fragment que nos permite agrupar una lista de elementos */
        <>
            <h1>{saludo}</h1>
            {/* <pre>{JSON.stringify(saludo, null,3)}</pre> */}
            <p>{subtitulo}</p>
        </>
    );

}


PrimeraApp.propTypes ={
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps ={
    subtitulo: "Soy un subtitulo"
}

export default PrimeraApp;