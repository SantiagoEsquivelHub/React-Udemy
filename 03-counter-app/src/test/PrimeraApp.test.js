import PrimeraApp from "../PrimeraApp";
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import React from 'react';


describe('Pruebas en PrimeraApp', () => {

    /* test('Debe mostrar el mensaje "Hola, soy Santi"', () => {

        const saludo = "Hola, soy Santi"
        const wrapper = render(<PrimeraApp saludo={saludo}/>)

        expect(wrapper.getByText(saludo)).toBeInTheDocument();
        
    }) */

    test('Debe mostrar el mensaje "Hola, soy Santi"', () => {

        const saludo = "Hola, soy Santi"
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>);
        
        expect(wrapper).toMatchSnapshot()
        
    })

})