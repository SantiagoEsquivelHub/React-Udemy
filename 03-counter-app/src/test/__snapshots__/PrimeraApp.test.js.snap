// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp Debe mostrar el mensaje "Hola, soy Santi" 1`] = `
<Fragment>
  <h1>
    Hola, soy Santi
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
